packages/runner/src/run.ts

```diff
@@ -223,16 +223,6 @@ export async function runTest(test: Test | Custom, runner: VitestRunner): Promis
           }
           await fn()
         }
-        // some async expect will be added to this array, in case user forget to await theme
-        if (test.promises) {
-          const result = await Promise.allSettled(test.promises)
-          const errors = result
-            .map(r => (r.status === 'rejected' ? r.reason : undefined))
-            .filter(Boolean)
-          if (errors.length) {
-            throw errors
-          }
```

Comment:
Previously it was waiting for `Promise.allSettled`, but I think we can simply use `Promise.all` since just throwing on first rejected promise would match with how test works.

Reply:
Not really, `failTask` will fail with every error, not just the first one

Reply:
Okay, I copied `Promise.allSettled`.